docs/source/cpp/arrays.rst

```diff
@@ -223,3 +227,53 @@ to some logical subsequence of the data.  This is done by calling the
 :func:`arrow::Array::Slice` and :func:`arrow::ChunkedArray::Slice` methods,
 respectively.
 
+.. _fromjson_helpers:
+
+FromJSON Helpers
+================
+
+A set of helper functions is provided for concisely creating Arrays and Scalars
+from JSON_ text. These helpers are intended to be used in examples, tests, or
+for quick prototyping and are not intended to be used where performance matters.
```

Comment:
Perhaps add a comment here that Arrow has a performant JSON reader that should be used in performance-sensitive cases? Perhaps adding a link too: https://arrow.apache.org/docs/cpp/json.html

Reply:
That's a good point. I added some text to this effect in 1bb08ebe57be18c2519d59f093b89a7a83b144e5.